src/d3d12/d3d12_model_pool.hpp

```diff
@@ -46,7 +52,9 @@ namespace wr
 		void Evict() final;
 		void MakeResident() final;
 
-		void StageMeshes(d3d12::CommandList* cmd_list);
+		void StageMeshes();
+
+		void WaitForStaging();
```

Comment:
If the render system executes the command lists it will wait for you there automatically since it gets staged at the beginning of the frame. We want to abstract all the gpu related code into the render system to make the renderer easy to use.